jumanji/environments/combinatorial/routing/rb_board_generator_v1.py

```diff
@@ -0,0 +1,319 @@
+#from constants import EMPTY, OBSTACLE, SOURCE, TARGET, HEAD, NOOP, LEFT, LEFT, UP, RIGHT, DOWN
+from constants import SOURCE, TARGET, HEAD
+from dataclasses import dataclass
+import numpy as np
+from copy import deepcopy
+import random
+WIRE = SOURCE
+
+@dataclass
+class Vector:
+    # Class of 2D tuple of ints, indicating the size of an array or a 2D vector in the array
+    x: int
+    y: int
+
+class Board:
+    """ The boards are 2D numpy arrays of wiring routes on a printed circuit board.
+
+    The coding of the boards is as follows:
+    Empty cells are coded 0
+    Obstacles are coded 1
+    Heads are encoded starting from 4 in multiples of 3: 4, 7, 10, ...
+    Targets are encoded starting from 3 in multiples of 3: 3, 6, 9, ...
+    Wiring routes connecting the head/target pairs are encoded starting
+	    at 2 in multiples of 3: 2, 5, 8, ...
+
+    ARGS:
+        x_dim, ydim (int, int) : Dimensions of the board
+
+    """
+    def __init__(self, x_dim: int, y_dim: int) -> None: 
+        self.layout = np.zeros((x_dim, y_dim), int)
+        self.dim = Vector(x_dim, y_dim)
+        self.num_wires = 0
+        return None
```

Comment:
I can accept the empty returns in other classes, but init really shouldn't have one in my opinion